lib/Runtime/Base/ThreadContext.cpp

```diff
@@ -4099,7 +4099,15 @@ BOOL ThreadContext::IsNativeAddress(void * pCodeAddr)
         boolean result;
         HRESULT hr = JITManager::GetJITManager()->IsNativeAddr(this->m_remoteThreadContextInfo, (intptr_t)pCodeAddr, &result);
         JITManager::HandleServerCallResult(hr, RemoteCallType::HeapQuery);
-        return result;
+
+        bool isNativeAddr = false;
+        for (Js::ScriptContext *scriptContext = scriptContextList; scriptContext && !isNativeAddr; scriptContext = scriptContext->next)
+        {
+            isNativeAddr = scriptContext->GetJitFuncRangeCache()->IsNativeAddr(pCodeAddr);
+        }
+
+        Assert(result == (isNativeAddr? 1:0));
```

Comment:
Should we pass scriptContext to IsNativeAddress and first check the current scriptContext before walking the list? I think most of calls should be from current scriptContext.